examples/imagenet/resnet50.py

```diff
@@ -0,0 +1,123 @@
+# Original author: yasunorikudo
+# (https://github.com/yasunorikudo/chainer-ResNet)
+
+import chainer
+import chainer.functions as F
+from chainer import initializers
+import chainer.links as L
+
+
+class BottleNeckA(chainer.Chain):
+
+    def __init__(self, in_size, ch, out_size, stride=2):
+        super(BottleNeckA, self).__init__(
+            conv1=L.Convolution2D(in_size, ch, 1, stride, 0,
+                                  initialW=initializers.HeNormal(),
+                                  nobias=True),
+            bn1=L.BatchNormalization(ch),
+            conv2=L.Convolution2D(ch, ch, 3, 1, 1,
+                                  initialW=initializers.HeNormal(),
+                                  nobias=True),
+            bn2=L.BatchNormalization(ch),
+            conv3=L.Convolution2D(ch, out_size, 1, 1, 0,
+                                  initialW=initializers.HeNormal(),
+                                  nobias=True),
+            bn3=L.BatchNormalization(out_size),
+
+            conv4=L.Convolution2D(in_size, out_size, 1, stride, 0,
+                                  initialW=initializers.HeNormal(),
+                                  nobias=True),
+            bn4=L.BatchNormalization(out_size),
+        )
+
+    def __call__(self, x):
+        h1 = F.relu(self.bn1(self.conv1(x)))
+        h1 = F.relu(self.bn2(self.conv2(h1)))
+        h1 = self.bn3(self.conv3(h1))
+        h2 = self.bn4(self.conv4(x))
+
+        return F.relu(h1 + h2)
+
+
+class BottleNeckB(chainer.Chain):
+
+    def __init__(self, in_size, ch):
+        super(BottleNeckB, self).__init__(
+            conv1=L.Convolution2D(in_size, ch, 1, 1, 0,
+                                  initialW=initializers.HeNormal(),
+                                  nobias=True),
+            bn1=L.BatchNormalization(ch),
+            conv2=L.Convolution2D(ch, ch, 3, 1, 1,
+                                  initialW=initializers.HeNormal(),
+                                  nobias=True),
+            bn2=L.BatchNormalization(ch),
+            conv3=L.Convolution2D(ch, in_size, 1, 1, 0,
+                                  initialW=initializers.HeNormal(),
+                                  nobias=True),
+            bn3=L.BatchNormalization(in_size),
+        )
+
+    def __call__(self, x):
+        h = F.relu(self.bn1(self.conv1(x)))
+        h = F.relu(self.bn2(self.conv2(h)))
+        h = self.bn3(self.conv3(h))
+
+        return F.relu(h + x)
+
+
+class Block(chainer.Chain):
```

Comment:
I feel it's better to inherit `ChainList` instead of `Chain`, and just pass each bottleneck child link to `add_link` one by one in `__init__`. Then, the `__call__` is also made simple.